packages/graph/hash_graph/lib/graph/src/api/error/mod.rs

```diff
@@ -0,0 +1,25 @@
+use std::collections::HashMap;
+
+use hash_status::Status as HashStatus;
+use serde::{Deserialize, Serialize};
+
+use crate::api::gen::status_payloads::{ErrorInfo, RequestInfo, ResourceInfo};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum StatusPayloads {
+    ErrorInfo(ErrorInfo),
+    RequestInfo(RequestInfo),
+    ResourceInfo(ResourceInfo),
+}
+
+impl ErrorInfo {
+    pub fn new(metadata: HashMap<String, Option<serde_json::Value>>, reason: String) -> Self {
```

Comment:
Will the reason ever be a non-static string? From the code below it seems it could be changed to `&'static str`

Reply:
The `ErrorInfo` type is generated with `String`, we don't control that